components/validator/guardian.go

```diff
@@ -91,16 +91,60 @@ func (g *Guardian) Stop() error {
 }
 
 func (g *Guardian) ValidateL2Output(ctx context.Context, outputRoot eth.Bytes32, l2BlockNumber uint64) (bool, error) {
+	g.log.Info("validating output...", "blockNumber", l2BlockNumber, "outputRoot", outputRoot)
 	localOutputRoot, err := g.outputRootAtBlock(ctx, l2BlockNumber)
 	if err != nil {
-		return false, fmt.Errorf("failed to get outputRootAtBlock: %w", err)
+		return false, fmt.Errorf("failed to get output root at block %d: %w", l2BlockNumber, err)
 	}
 	isValid := bytes.Equal(outputRoot[:], localOutputRoot[:])
 	return isValid, nil
 }
 
 func (g *Guardian) ConfirmTransaction(ctx context.Context, transactionId *big.Int) (*types.Transaction, error) {
-	txOpts := utils.NewSimpleTxOpts(ctx, g.cfg.TxManager.From(), g.cfg.TxManager.Signer)
+	cCtx, cCancel := context.WithTimeout(ctx, g.cfg.NetworkTimeout)
+	executionTx, err := g.securityCouncilContract.Transactions(&bind.CallOpts{Context: cCtx}, transactionId)
+	cCancel()
+	if err != nil {
+		return nil, err
+	}
+	if executionTx.Executed {
+		return nil, fmt.Errorf("transaction %d already executed", transactionId)
```

Comment:
If the tx was already executed, it would be nice to return err = nil to terminate the loop.
And I think we can conditionally avoid sending a sendTrasaction for a tx that is nil.
```suggestion
		return nil, nil
```